presto-parser/src/main/java/com/facebook/presto/sql/tree/GroupingSets.java

```diff
@@ -64,13 +66,13 @@ public boolean equals(Object o)
             return false;
         }
         GroupingSets groupingSets = (GroupingSets) o;
-        return Objects.equals(sets, groupingSets.sets);
+        return Objects.equals(ImmutableSet.copyOf(sets), ImmutableSet.copyOf(groupingSets.sets));
```

Comment:
This seems to be an irrelevant bugfix. Pull this in a separate commit?

Additionally, can we make `sets` set of set instead? Or maybe set of list?